js-OOP/spreadAndRest.js

```diff
@@ -0,0 +1,168 @@
+// Task 1
+
+const array1 = [1, 2, 3];
+const array2 = [4, 5, 6];
+
+const array3 = [...array1, ...array2];
+
+console.log(array3);
+
+//Task 2
+
+function multiply(multiplier, ...arguments) {
+    const result = arguments.map(item => item * multiplier);
+    return result;
+}
+
+const result = multiply(77, 2, 3, 6);
+console.log(result);
+
+//Task 3
+
+class ProductClass {
+    constructor(name, price, quantity) {
+        this.name = name;
+        this.price = price;
+        this.quantity = quantity;
+    }
+
+    getTotalPrice() {
+        return this.price * this.quantity;
+    }
+
+    purchase(quantity) {
+        if (quantity <= this.quantity && quantity > 0) {
+            this.quantity -= quantity;
+            console.log(`Покупка ${quantity} одиниць товару "${this.name}" виконана.`);
+        } else {
+            console.log(`Неможливо виконати покупку для товару "${this.name}".`);
+        }
+    }
+}
+
+const tShirt = new ProductClass("Футболка", 50, 20);
+const jeans = new ProductClass("Джинси", 90, 34);
+const hat = new ProductClass("Кепка", 5, 130);
+
+console.log(`Загальна вартість футболок: $${tShirt.getTotalPrice()}`);
+console.log(`Загальна вартість джинсів: $${jeans.getTotalPrice()}`);
+console.log(`Загальна вартість кепок: $${hat.getTotalPrice()}`);
+
+tShirt.purchase(5);
+console.log(`Залишок футболок на складі: ${tShirt.quantity}`);
+
+jeans.purchase(10);
+console.log(`Залишок джинсів на складі: ${jeans.quantity}`);
+
+hat.purchase(20);
+console.log(`Залишок кепок на складі: ${hat.quantity}`);
+
+//Task 4
+
+class CarClass {
+    constructor(make, model, year) {
+        this.make = make;
+        this.model = model;
+        this.year = year;
+        this.speed = 0;
+    }
```

Comment:
speed property should be private
```suggestion
#speed
    constructor(make, model, year) {
        this.make = make;
        this.model = model;
        this.year = year;
        this.#speed = 0;
    }
```